WalletWasabi.Packager/MacSignTools.cs

```diff
@@ -458,11 +465,38 @@ private static void SignDirectory(string[] files, string workingDir, string sign
 			}
 		}
 
-		private static string[] GetExecutables(string workingDir)
+		private static IEnumerable<string> GetExecutables(string appPath)
+		{
+			string result = ExecuteBashCommand($"find -H \"{appPath}\" -print0 | xargs -0 file | grep \"Mach-O.* executable\"");
+
+			var lines = result.Split("\n").Where(x => !string.IsNullOrWhiteSpace(x));
+			var files = lines.Select(line => line.Split(":").First());
+
+			return files;
+		}
+
+		static string ExecuteBashCommand(string command)
 		{
-			// Can be implemented as: find -H YourAppBundle -print0 | xargs -0 file | grep "Mach-O .*executable"
+			// according to: https://stackoverflow.com/a/15262019/637142
+			// thans to this we will pass everything as one command
+			command = command.Replace("\"", "\"\"");
+
+			using var process = Process.Start(new ProcessStartInfo
+			{
+				FileName = "/bin/bash",
+				Arguments = $"-c \"{command}\"",
+				UseShellExecute = false,
+				RedirectStandardError = true,
+				RedirectStandardOutput = true,
+				CreateNoWindow = true
+			});
+
+			var reader = process.StandardOutput.ReadToEndAsync();
```

Comment:
Isn't it safer to use: `string result = await process.StandardOutput.ReadToEndAsync();` rather than starting that task and using `process.WaitForExit()` then? I mean, is it somehow guaranteed that it's allowed use of the `Process` API? 

It looks to me that you can use `await` if you change only a few method signatures as Packager.Program.Main is async already.

Reply:
My intention here is to avoid the "pipe deadlock". I removed async and make sure to read before the waitforexit. 

> A deadlock condition can result if the parent process calls p.WaitForExit before p.StandardOutput.ReadToEnd and the child process writes enough text to fill the redirected stream.

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=netcore-3.1